orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -167,6 +169,27 @@ TEST(GradientGraphBuilderTest, BuildGradientGraphTest) {
   }
 }
 
+TEST(GradientGraphBuilderTest, BuildConcatGradientGraphTest) {
+  const auto config = MakeBasicTrainingConfig();
+  PathString backprop_model_file;
+  ASSERT_STATUS_OK(BuildBackPropGraph(CONCAT_MODEL_PATH, config, backprop_model_file));
```

Comment:
onnxruntime_test_all.exe!gsl::not_null<onnxruntime::Graph *>::get() Line 1279 C++
  onnxruntime_test_all.exe!gsl::not_null<onnxruntime::Graph *>::operator*() Line 1287 C++
  onnxruntime_test_all.exe!onnxruntime::GraphTransformer::Recurse(onnxruntime::Node & node, bool & modified, int graph_level, const onnxruntime::logging::Logger & logger) Line 47 C++
  onnxruntime_test_all.exe!onnxruntime::RuleBasedGraphTransformer::ApplyImpl(onnxruntime::Graph & graph, bool & modified, int graph_level, const onnxruntime::logging::Logger & logger) Line 84 C++
  onnxruntime_test_all.exe!onnxruntime::GraphTransformer::Apply(onnxruntime::Graph & graph, bool & modified, const onnxruntime::logging::Logger & logger) Line 14 C++
  onnxruntime_test_all.exe!onnxruntime::GraphTransformerManager::ApplyTransformers(onnxruntime::Graph & graph, onnxruntime::TransformerLevel level, const onnxruntime::logging::Logger & logger) Line 32 C++
  onnxruntime_test_all.exe!onnxruntime::training::TrainingSession::ApplyTransformationsToMainGraph(const std::unordered_set<std::string,std::hash<std::string>,std::equal_to<std::string>,std::allocator<std::string>> & weights_to_train, bool enable_gelu_approximation) Line 495 C++
  onnxruntime_test_all.exe!onnxruntime::training::TrainingSession::ConfigureForTraining(const onnxruntime::training::TrainingSession::TrainingConfiguration & config, onnxruntime::training::TrainingSession::TrainingConfigurationResult & config_result_out) Line 218 C++
  onnxruntime_test_all.exe!onnxruntime::test::BuildBackPropGraph(const std::wstring & forward_model_file, const onnxruntime::training::TrainingSession::TrainingConfiguration & config, std::wstring & backward_model_file) Line 70 C++
  onnxruntime_test_all.exe!onnxruntime::test::GradientGraphBuilderTest_BuildConcatGradientGraphTest_Test::TestBody() Line 175 C++

Reply:
failed on windows

---
In reply to: [463407164](https://github.com/microsoft/onnxruntime/pull/4675#discussion_r463407164) [](ancestors = 463407164)